subsys/shell/modules/zyclictest_service.c

```diff
@@ -0,0 +1,250 @@
+	if (ret) {
+		LOG_ERR("Error while setting zyclictest thread name: %d", ret);
+	}
+
+	return 0;
+}
+
+int zyclictest_exit(void)
+{
+	int ret;
+
+	zyc_running = 0;
+
+	ret = k_thread_join(zyc_tid, K_FOREVER);
+	if (ret) {
+		LOG_ERR("Error while terminating zyclictest thread: %d", ret);
+	}
+
+	zyc_tid = NULL;
+
+	k_timer_stop(&zyc_timer);
+
+	return 0;
+}
+
+static int cmd_zyclictest_start(const struct shell *sh, size_t argc, char **argv)
+{
+	int ret;
+	int rv, err = 0;
+
+#ifndef CONFIG_NATIVE_LIBC
+	getopt_init();
+#endif
+
+	zyc_data.thr_prio = -CONFIG_NUM_COOP_PRIORITIES;
+	zyc_data.interval_us = 1000000LL;
+	zyc_data.cnt_lat = 0;
+	memset(zyc_data.max_lat, 0, sizeof(zyc_data.max_lat));
+	memset(zyc_data.cnt_err, 0, sizeof(zyc_data.cnt_err));
+	memset(zyc_data.cnt_ovl, 0, sizeof(zyc_data.cnt_ovl));
+	memset(zyc_data.lat, 0, sizeof(zyc_data.lat));
+
+	while ((rv = getopt(argc, argv, "i:p:")) != -1) {
+		switch (rv) {
+		case 'i':
+			zyc_data.interval_us = shell_strtoull(optarg, 0, &err);
+			if (err) {
+				shell_error(sh, "invalid interval '%s' err: %d", optarg, err);
+				return -EINVAL;
+			}
+			break;
+		case 'p':
+			zyc_data.thr_prio = shell_strtol(optarg, 0, &err);
+			if ((err) ||
+			    (zyc_data.thr_prio < -CONFIG_NUM_COOP_PRIORITIES) ||
+			    (zyc_data.thr_prio > CONFIG_NUM_PREEMPT_PRIORITIES)) {
+				shell_error(sh, "invalid priority '%s' err: %d", optarg, err);
+				return -EINVAL;
+			}
+			break;
+		default:
+			return -EINVAL;
+		}
+	}
+
+	shell_print(sh, "Cycle interval: %llu us", zyc_data.interval_us);
+	shell_print(sh, "Priority: %d", zyc_data.thr_prio);
+
+	ret = zyclictest_init();
+	if (ret) {
+		LOG_ERR("Error while initializing zyclictest: %d", ret);
+	}
+
+	return 0;
+}
+
+static int cmd_zyclictest_stop(const struct shell *sh, size_t argc, char **argv)
+{
+	int i;
+	int max_lat;
+
+	zyclictest_exit();
+
+	shell_print(sh, "Count: %d", zyc_data.cnt_lat);
+	shell_print(sh, "            \t   IRQ\tThread\n");
+	shell_print(sh, "Max-Latency:\t%6d\t%6d",
+					zyc_data.max_lat[ZYC_INT], zyc_data.max_lat[ZYC_THR]);
+	shell_print(sh, "Errors:     \t%6d\t%6d",
+					zyc_data.cnt_err[ZYC_INT], zyc_data.cnt_err[ZYC_THR]);
+	shell_print(sh, "Overflow:   \t%6d\t%6d",
+					zyc_data.cnt_ovl[ZYC_INT], zyc_data.cnt_ovl[ZYC_THR]);
+	shell_print(sh, "Histogram:");
+
+	max_lat = MIN(zyc_data.max_lat[ZYC_THR], ZYC_MAX_HIST - 1);
+	for (i = 0; i <= max_lat; i++) {
+		shell_print(sh, "%3d         \t%6d\t%6d",
+				i,
+				zyc_data.lat[ZYC_INT][i],
+				zyc_data.lat[ZYC_THR][i]);
+	}
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_zyclictest,
+	SHELL_CMD_ARG(start, NULL, "Start Zyclictest\n"
+		"Usage:\n"
+		"zyclictest [-i <interval-us>] [-p <prio>]",
+		cmd_zyclictest_start, 1, 4),
+	SHELL_CMD(stop, NULL, "Stop Zyclictest.", cmd_zyclictest_stop),
+	SHELL_SUBCMD_SET_END /* Array terminated. */
+);
+
+SHELL_CMD_REGISTER(zyclictest, &sub_zyclictest, "Zephyr cyclictest latency measurement", NULL);
```

Comment:
do you forsee more subcommands to be creaed in addition to "start"? otherwise I guess it may make sense to just have a single `zyclictest` top level shell command